### PR TITLE
add macro for calculating radians

### PR DESCRIPTION
**Pull request by Pespiri**

### CODE REVIEW

**Comment by TheDevMinerTV** [comment]
What exactly will this solve?

**Comment by Pespiri** [comment]
> What exactly will this solve?

adds the ability to specify weird sensor mounting angles on custom setups

**Review by unlogisch04 — APPROVED** [review]
Results are the same and makes special angles better handled in code.

**Comment by ImUrX** [comment]
Π

File: src/consts.h [hunk 1]
```diff
@@ -82,10 +82,13 @@
...
-#define DEG_90 -PI / 2
-#define DEG_180 PI
-#define DEG_270 PI / 2
+// Get radian for a given angle from 0° to 360° (2*PI*r, solve for r given an angle, range -180° to 180°)
+#define DEG_X(deg) (float)((((deg) < 180.0f ? 0 : 360.0f) - (deg)) * PI / 180.0f)
+
+#define DEG_0 DEG_X(0.0f)
+#define DEG_90 DEG_X(90.0f)
+#define DEG_180 DEG_X(180.0f)
+#define DEG_270 DEG_X(270.0f)
```
Comment (unlogisch04) [inline src/consts.h:91]:
```suggestion
#define DEG_0 0.f
#define DEG_90 -PI / 2
#define DEG_180 PI
#define DEG_270 PI / 2
```
The Values of the result are not the same. So it is preferable to stay with the existing formula. for the defined constant
```
[INFO ] [SensorManager]    DEG_0  : 0.00000000000000000000
[INFO ] [SensorManager] PI_DEG_0  : 0.00000000000000000000
[INFO ] [SensorManager]    DEG_90 : -1.57079637050628662109
[INFO ] [SensorManager] PI_DEG_90 : -1.57079632679489655800
[INFO ] [SensorManager]    DEG_180: 3.14159274101257324219
[INFO ] [SensorManager] PI_DEG_180: 3.14159265358979311600
[INFO ] [SensorManager]    DEG_270: 1.57079637050628662109
[INFO ] [SensorManager] PI_DEG_270: 1.57079632679489655800
```

**Review by unlogisch04 — CHANGES_REQUESTED** [review]

File: src/consts.h (hunk 1 above)
Comment (Pespiri) [reply src/consts.h:91]: this is caused by comparing a double to a float.
the macro is cast as float, while the previous formula was automatically cast as double.
I'll fix the macro instead 😄

Comment (Pespiri) [reply src/consts.h:91]: Macro with cast to float
image

Macro with cast to double
image

**Comment by Pespiri** [comment]
@unlogisch04 float cast issue has been corrected, math be mathing 😋

**Review by unlogisch04 — APPROVED** [review]
LGTM

**Review by Eirenliel — APPROVED** [review]

**Pull request merged**